metrics/internal/collectors/health_score.go

```diff
@@ -0,0 +1,302 @@
+package collectors
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	prometheusconfig "github.com/prometheus/common/config"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/red-hat-storage/ocs-operator/metrics/v4/internal/options"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	healthScorePrometheusRuleName = "odf-health-alerts"
+	alertManagerClientTimeout     = 10 * time.Second
```

Comment:
How have we added this timeouts? I am just a default random value or is their anything specific?

Reply:
Chose 10s because the in-cluster Alertmanager API is usually fast, but we need to bound the scrape so it can’t hang a Prometheus collect